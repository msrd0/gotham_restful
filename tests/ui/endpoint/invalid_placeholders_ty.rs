#[macro_use]
extern crate gotham_restful;
use gotham_restful::gotham::hyper::Method;

#[derive(Resource)]
#[resource(endpoint)]
struct FooResource;

#[derive(Debug)]
struct FooPlaceholders {
	foo: String
}

#[endpoint(method = "Method::GET", uri = ":foo")]
fn endpoint(_: FooPlaceholders) {
	unimplemented!()
}

fn main() {}
