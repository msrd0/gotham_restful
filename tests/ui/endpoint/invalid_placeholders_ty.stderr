error[E0277]: the trait bound `FooPlaceholders: OpenapiType` is not satisfied
  --> tests/ui/endpoint/invalid_placeholders_ty.rs:15:16
   |
15 | fn endpoint(_: FooPlaceholders) {
   |                ^^^^^^^^^^^^^^^ the trait `OpenapiType` is not implemented for `FooPlaceholders`
   |
   = help: the following other types implement trait `OpenapiType`:
             &'a T
             ()
             BTreeMap<K, V>
             BTreeSet<T>
             CStr
             CString
             HashMap<K, V, S>
             HashSet<T, S>
           and 36 others
note: required by a bound in `gotham_restful::EndpointWithSchema::Placeholders`
  --> src/endpoint.rs
   |
   |     #[openapi_bound("Placeholders: OpenapiType")]
   |                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `gotham_restful::EndpointWithSchema::Placeholders`

error[E0277]: the trait bound `FooPlaceholders: Clone` is not satisfied
  --> tests/ui/endpoint/invalid_placeholders_ty.rs:15:16
   |
15 | fn endpoint(_: FooPlaceholders) {
   |                ^^^^^^^^^^^^^^^ the trait `Clone` is not implemented for `FooPlaceholders`
   |
note: required by a bound in `gotham_restful::EndpointWithSchema::Placeholders`
  --> src/endpoint.rs
   |
   |     type Placeholders: PathExtractor<Body> + Clone + Sync;
   |                                              ^^^^^ required by this bound in `gotham_restful::EndpointWithSchema::Placeholders`
help: consider annotating `FooPlaceholders` with `#[derive(Clone)]`
   |
10 | #[derive(Clone)]
   |

error[E0277]: the trait bound `FooPlaceholders: StateData` is not satisfied
  --> tests/ui/endpoint/invalid_placeholders_ty.rs:15:16
   |
15 | fn endpoint(_: FooPlaceholders) {
   |                ^^^^^^^^^^^^^^^ the trait `StateData` is not implemented for `FooPlaceholders`
   |
   = help: the following other types implement trait `StateData`:
             AuthSource
             AuthStatus<T>
             Body
             CorsConfig
             FilePathExtractor
             HeaderMap
             IdPlaceholder<ID>
             Method
           and 9 others
note: required by a bound in `gotham_restful::EndpointWithSchema::Placeholders`
  --> src/endpoint.rs
   |
   |     type Placeholders: PathExtractor<Body> + Clone + Sync;
   |                        ^^^^^^^^^^^^^^^^^^^ required by this bound in `gotham_restful::EndpointWithSchema::Placeholders`

error[E0277]: the trait bound `FooPlaceholders: StaticResponseExtender` is not satisfied
  --> tests/ui/endpoint/invalid_placeholders_ty.rs:15:16
   |
15 | fn endpoint(_: FooPlaceholders) {
   |                ^^^^^^^^^^^^^^^ the trait `StaticResponseExtender` is not implemented for `FooPlaceholders`
   |
   = help: the following other types implement trait `StaticResponseExtender`:
             FilePathExtractor
             IdPlaceholder<ID>
             NoopExtractor
             NoopPathExtractor
             NoopQueryStringExtractor
             NoopResponseExtender
note: required by a bound in `gotham_restful::EndpointWithSchema::Placeholders`
  --> src/endpoint.rs
   |
   |     type Placeholders: PathExtractor<Body> + Clone + Sync;
   |                        ^^^^^^^^^^^^^^^^^^^ required by this bound in `gotham_restful::EndpointWithSchema::Placeholders`

error[E0277]: the trait bound `for<'de> FooPlaceholders: serde::de::Deserialize<'de>` is not satisfied
  --> tests/ui/endpoint/invalid_placeholders_ty.rs:15:16
   |
15 | fn endpoint(_: FooPlaceholders) {
   |                ^^^^^^^^^^^^^^^ the trait `for<'de> serde::de::Deserialize<'de>` is not implemented for `FooPlaceholders`
   |
   = help: the following other types implement trait `serde::de::Deserialize<'de>`:
             &'a Path
             &'a [u8]
             &'a str
             ()
             (T0, T1)
             (T0, T1, T2)
             (T0, T1, T2, T3)
             (T0, T1, T2, T3, T4)
           and 202 others
note: required by a bound in `gotham_restful::EndpointWithSchema::Placeholders`
  --> src/endpoint.rs
   |
   |     type Placeholders: PathExtractor<Body> + Clone + Sync;
   |                        ^^^^^^^^^^^^^^^^^^^ required by this bound in `gotham_restful::EndpointWithSchema::Placeholders`
